AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/Model/keyrings.smithy

```diff
@@ -340,7 +341,10 @@ structure CreateAwsKmsHierarchicalKeyringInput {
     ttlSeconds: PositiveLong,
 
     @javadoc("Which type of local cache to use.")
-    cache : CacheType
+    cache : CacheType,
+
+    @documentation("Shared cache across multiple Hierarchical Keyrings. For every Hierarchical Keyring, one out the `cache` or `sharedCache` parameter MUST be set, not both. If both parameters are set, an exception will be thrown. If neither of the two parameters are set, a DefaultCache is initialized to be used with the Hierarchical Keyring with entryCapacity = 1000.")
```

Comment:
```suggestion
    @documentation("Sets the cache to which this Hierarchical Keyring will be bound. Either the `cache` or `sharedCache` parameter may be set, not both. If both parameters are set, an exception will be thrown. If neither of the two parameters are set, a DefaultCache is initialized to be used with the only this Hierarchical Keyring with entryCapacity = 1000.")
```

Reply:
I am going to leave this open because I'm trying out (and opening another PR for) the other implementation (without the extra `sharedCache` parameter) to see if that's better. Will resolve your comments about documentation if we switch back to this implementation.